client/components/date-range/index.js

```diff
@@ -0,0 +1,526 @@
+				// this components state
+				this.props.onDateSelect( this.state.startDate, this.state.endDate );
+			}
+		);
+	}
+
+	/**
+	 * Updates the "stale" data to reflect the current start/end dates
+	 * This causes any cached data to be lost and thus the current start/end
+	 * dates are persisted. Typically called when user clicks "Apply"
+	 */
+	commitDates() {
+		this.setState(
+			previousState => ( {
+				staleStartDate: previousState.startDate, // update cached stale dates
+				staleEndDate: previousState.endDate, // update cached stale dates
+				staleDatesSaved: false,
+			} ),
+			() => {
+				this.props.onDateCommit( this.state.startDate, this.state.endDate );
+				this.closePopover();
+			}
+		);
+	}
+
+	/**
+	 * Reverts current start/end dates to the cache "stale" versions
+	 * Typically required when the user makes a selection but then dismisses
+	 * the DateRange without clicking "Apply"
+	 */
+	revertDates() {
+		this.setState( previousState => {
+			const newState = { staleDatesSaved: false };
+
+			if ( previousState.staleStartDate && previousState.staleEndDate ) {
+				newState.startDate = previousState.staleStartDate;
+				newState.endDate = previousState.staleEndDate;
+				newState.textInputStartDate = this.formatDateToLocale( this.state.staleStartDate );
+				newState.textInputEndDate = this.formatDateToLocale( this.state.staleEndDate );
+			}
+
+			return newState;
+		} );
+	}
+
+	/**
+	 * Converts a moment date to a native JS Date object
+	 * @param  {MomentJSDate} momentDate a momentjs date object to convert
+	 * @return {DATE}            the converted JS Date object
+	 */
+	momentDateToJsDate( momentDate ) {
+		return this.props.moment.isMoment( momentDate ) ? momentDate.toDate() : momentDate;
+	}
+
+	/**
+	 * Converts a native JS Date object to a MomentJS Date object
+	 * @param  {Date} nativeDate date to be converted
+	 * @return {MomentJSDate}            the converted Date
+	 */
+	nativeDateToMoment( nativeDate ) {
+		return this.props.moment( nativeDate );
+	}
+
+	/**
+	 * Formats a given date to the appropriate format for the
+	 * current locale
+	 * @param  {Date|MomentJSDate} date the date to be converted
+	 * @return {String}      the date as a formatted locale string
+	 */
+	formatDateToLocale( date ) {
+		return this.props.moment( date ).format( 'L' );
+	}
+
+	/**
+	 * 	Gets the locale appropriate date format (eg: "MM/DD/YYYY")
+	 * @return {String} date format as a string
+	 */
+	getLocaleDateFormat() {
+		return this.props.moment.localeData().longDateFormat( 'L' );
+	}
+
+	/**
+	 * Enforces that given date is within the bounds of the
+	 * range specified
+	 * @param  {Moment} date             momentJS instance
+	 * @param  {Moment|Date} options.dateFrom the start of the date range
+	 * @param  {Moment|Date} options.dateTo   the end of the date range
+	 * @return {Moment}                  the date clamped to be within the range
+	 */
+	clampDateToRange( date, { dateFrom, dateTo } ) {
```

Comment:
`clampDateToRange` looks like it could live as a standalone util function as it doesn't seem to benefit from being a member of the component.
That said, this definitely isn't a blocker as there's also no harm in it being a member method.

Reply:
Will have a 🤔